platforms/core-execution/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorErrorHandlingIntegrationTest.groovy

```diff
@@ -148,8 +147,11 @@ class WorkerExecutorErrorHandlingIntegrationTest extends AbstractWorkerExecutorI
 
             task runInWorker(type: WorkerTask) {
                 isolationMode = $isolationMode
-                foo = new FooWithUnserializableBar()
                 finalizedBy runAgainInWorker
```

Comment:
🤔 

Weirdly, I got no build error when I tried this locally, so the test failed. Is there another change that needs to go with this?

Also, could the test just below benefit from a similar fix? 

https://github.com/gradle/gradle/blob/d75fbddb421ff875b1b2e349662c9802afb04e32/platforms/core-execution/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorErrorHandlingIntegrationTest.groovy#L176